### PR TITLE
Add with properties to override data

### PR DESCRIPTION
**Pull request by andidev**
Adds with properties to be able to set specific data to be non random
e.g.
```java
fairy.person(withFirstName("Jimi"), withLastName("Hendrix"), male())
```

generates
```
  address=57543 San Francisco
  firstName=Jimi
  middleName=
  lastName=Hendrix
  email=hendrix@mail.com
  username=jimih
  password=fbpXO9ZB
  sex=MALE
  telephoneNumber=116-871-4540
  dateOfBirth=1998-05-14T09:08:26.663+02:00
  age=18
  company=io.codearte.jfairy.producer.company.Company@1e12a5a6
  companyEmail=jimi.hendrix@memortechlimited.com
  nationalIdentityCardNumber=822-21-9118
  nationalIdentificationNumber=
  passportNumber=zj3b0pBFG
```

This PR adds with properties to override any of the fields on the Company and the Person class.

### CODE REVIEW

File: src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java [hunk 1]
```diff
@@ -15,30 +16,68 @@
...
 		this.vatIdentificationNumberProvider = vatIdentificationNumberProvider;
+
+		for (CompanyProperties.CompanyProperty companyProperty : companyProperties) {
+			companyProperty.apply(this);
+		}
 	}
 
 
 	@Override
 	public Company get() {
```
Comment (OlgaMaciaszek) [inline src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java]: imho there are too many conditions in this method and it's too large; maybe you could extract smaller methods and substitute null checks in conditions with elvis operators in return statements where possible?
Please also apply this to other similar places in this PR.

File: src/test/groovy/io/codearte/jfairy/producer/person/PersonSpec.groovy [hunk 2]
```diff
@@ -214,5 +231,118 @@ class PersonSpec extends Specification {
 			person.passportNumber()
 	}
 
+	def "withFirstName should create person with specific first name"() {
+		when:
+			def person = fairy.person(withFirstName("Specificfirstname"))
```
Comment (OlgaMaciaszek) [inline src/test/groovy/io/codearte/jfairy/producer/person/PersonSpec.groovy]: Please use type in place of `def` where possible.
Please also apply this to other similar places in this PR.

File: src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java [hunk 3]
```diff
@@ -86,31 +100,66 @@ public Person get() {
...
-		String companyEmail = stripAccents(lowerCase(firstName + '.' + lastName + '@' + company.domain()));
-		// FIXME: Replace this with baseProducer
-		String password = RandomStringUtils.randomAlphanumeric(8);
 
-		return new Person(firstName, middleName, lastName, addressProvider.get(), email,
+		if (companyEmail == null) {
+			companyEmail = stripAccents(lowerCase(firstName + '.' + lastName + '@' + company.domain()));
+		}
+		if (password == null) {
+			// FIXME: Replace this with baseProducer
```
Comment (OlgaMaciaszek) [inline src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java]: I would avoid FIXME items in master. It would be better to implement the final version here, as it should not be that complex.

Comment (andidev) [inline src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java]: This fixme was there before I added my changes. 
Is it okey to remove this fixme? Or should I add the password generation to the baseProducer?

File: src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java (hunk 1 above)
Comment (andidev) [inline src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java]: I'll do that

File: src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java (hunk 3 above)
Comment (OlgaMaciaszek) [inline src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java]: Sorry. Haven't realised it was before. If you can, add the password generation to baseProducer, please; but if you don't want to work on that right now, please let me know - will just merge the current PR as is once the other minor issues are fixed.

Comment (andidev) [inline src/main/java/io/codearte/jfairy/producer/person/PersonProvider.java]: I can do it though I think it would be better to fix this in another PR. It would be nice to have #68 merged or closed before further changes since it may cause lots of merge conflict otherwise.

File: src/test/groovy/io/codearte/jfairy/producer/person/PersonSpec.groovy (hunk 2 above)
Comment (andidev) [inline src/test/groovy/io/codearte/jfairy/producer/person/PersonSpec.groovy]: I would like to suggest to fix the `def` issue in another PR after PR #68 is merged or closed. This is to avoid merge conflicts. I can do the fix asap as soon as a decision is made.

Comment (OlgaMaciaszek) [inline src/test/groovy/io/codearte/jfairy/producer/person/PersonSpec.groovy]: Ok. Let's do it this way. I will merge it now. Please make another PR with def changes. Could you also include some instances where that has already been done.

Comment (OlgaMaciaszek) [inline src/test/groovy/io/codearte/jfairy/producer/person/PersonSpec.groovy]: I know this is quite a subjective and opinionated approach when it comes to those `def`, but as in the Codearte team we all seem to share it, we would like to keep it this way.

File: src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java (hunk 1 above)
Comment (andidev) [inline src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java]: @OlgaMaciaszek
I did not commit the suggested changes mentioned in the above comment. Should I do them and do another pull requests?

Comment (OlgaMaciaszek) [inline src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java]: Yes, please. :)
